Add doc for yield clause (#347)
kuzudb/kuzu-docs: src/content/docs/cypher/query-clauses/call.md

CALL table_info('student')
YIELD `property id` as student_id,  name as student_name, type as student_type, `default expression` as student_default, `primary key` as student_pk 
RETURN *;
```

Result:
```
┌───────────┬─────────────┬─────────────┬────────────────┬───────────┬────────────┬──────────────┬──────────────┬─────────────────┬────────────┐
│ person_id │ person_name │ person_type │ person_default │ person_pk │ student_id │ student_name │ student_type │ student_default │ student_pk │
│ INT32     │ STRING      │ STRING      │ STRING         │ BOOL      │ INT32      │ STRING       │ STRING       │ STRING          │ BOOL       │
├───────────┼─────────────┼─────────────┼────────────────┼───────────┼────────────┼──────────────┼──────────────┼─────────────────┼────────────┤
│ 0         │ id          │ INT64       │ NULL           │ True      │ 0          │ id           │ INT64        │ NULL            │ True       │
└───────────┴─────────────┴─────────────┴────────────────┴───────────┴────────────┴──────────────┴──────────────┴─────────────────┴────────────┘
```

:::caution[Note]

> 1. If the yield clause is used after a call function, all return columns of the function must appear in the yield clause.

Reviewer: ```suggestion
1. If the `YIELD` clause is used after a `CALL` function, **all** return columns of the function must appear in the `YIELD` clause.
```

For example:
```
CALL table_info('person')
YIELD `property id` as person_id
RETURN person_id
```
The query throws an exception since not all returns columns of the `table_info` function appear in the yield clause.

> 2. The column names to yield must match the origianl return column names of the call function.

Reviewer: ```suggestion
2. The column names to yield must match the original return column names of the call function.
```

For example:
```
CALL current_setting('threads')
YIELD thread as threads_num
RETURN *;
```
The query throws an exception since the column name to yield is `thread` which doesn't match the return column name(`threads`) of the call function.

> 3. Unlike neo4j, `Yield` clause must be followed by a return clause and `yield *` is not allowed in kuzu.

Reviewer: ```suggestion
3. The syntax in Kùzu Cypher is different from other systems like Neo4j. In Kùzu, the `YIELD` clause must be followed by a return clause. `YIELD *` is not allowed in Kùzu.
```
